### PR TITLE
[lcn] Add support for commands from LCN to openHAB

### PR DESCRIPTION
**Pull request by fwolter**
Signed-off-by: Fabian Wolter <github@fabian-wolter.de>

### CODE REVIEW

File: bundles/org.openhab.binding.lcn/src/main/resources/ESH-INF/thing/thing-types.xml [hunk 1]
```diff
@@ -658,4 +659,17 @@
...
+		<label>Host Command (to openHAB)</label>
+		<channels>
+			<channel typeId="sendKeys" id="sendKeys"/>
+		</channels>
+	</channel-group-type>
+
+	<channel-type id="sendKeys" advanced="true">
+		<kind>trigger</kind>
+		<label>Send Keys</label>
+		<event/>
```
Comment (cpmeister) [inline bundles/org.openhab.binding.lcn/src/main/resources/ESH-INF/thing/thing-types.xml:673]: Is it necessary to include `<event/>`?

Comment (fwolter) [reply bundles/org.openhab.binding.lcn/src/main/resources/ESH-INF/thing/thing-types.xml]: I think so, as the trigger contains payload like "A1:HIT" or "B2:MAKE". I added the options tag.

File: bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleHostCommandSubHandler.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+        }
+
+        // PCHK 3.22 supports only the old 'send key' command with key tables A-C
+        for (int keyTableNumber = 0; keyTableNumber < LcnDefs.KEY_TABLE_COUNT_UNTIL_0C030C0; keyTableNumber++) {
+            String keyTableName = LcnDefs.KeyTable.values()[keyTableNumber].name();
+
+            for (int keyNumber = 0; keyNumber < LcnDefs.KEY_COUNT; keyNumber++) {
+                int actionRaw = (keyTableAndActionMask >> (keyTableNumber * 2)) & 3;
+
+                if (actionRaw > LcnDefs.SendKeyCommand.DONTSEND.ordinal()
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.lcn/src/main/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleHostCommandSubHandler.java:62]: It looks lik the code depends on the ordering in the enum, which is dangerous. Maybe add an int to the enum. Or in some other way store the range used here in some variable.

File: bundles/org.openhab.binding.lcn/src/test/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleHostCommandSubHandlerTest.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+import org.openhab.binding.lcn.internal.common.LcnChannelGroup;
+
+/**
+ * Test class.
+ *
+ * @author Fabian Wolter - Initial contribution
+ */
+@NonNullByDefault
+public class LcnModuleHostCommandSubHandlerTest extends AbstractTestLcnModuleSubHandler {
+    private @NonNullByDefault({}) LcnModuleHostCommandSubHandler l;
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.lcn/src/test/java/org/openhab/binding/lcn/internal/subhandler/LcnModuleHostCommandSubHandlerTest.java:29]: L is very easily mixed with 1. Maybe use a different, longer name.

**Review by Hilbrand — CHANGES_REQUESTED** [review]

**Review by Hilbrand — APPROVED** [review]
LGTM

**Pull request merged**